File: apm287-stock-mobile/src/common/StockInfo.ts
export class StockInfo {
    code: string;
    startingPrice: number;
	price: number;
	change: number;
	holding: number;

    constructor(code: string, startingPrice: number, price: number, change: number, holding: number) {
        this.code = code;
        this.startingPrice = startingPrice;
	    this.price = price;
	    this.change = change;
	    this.holding = holding;
    }

}

File: apm287-stock-backend/src/main/java/de/novatec/apm287/common/StockInfo.java
package de.novatec.apm287.common;

import java.util.Collection;
import java.util.Collections;

public class StockInfo {
	public String code;
	public final double startingPrice;
	public double price;
	public double change;
	public int holding;
	
	public StockInfo(String code, double price) {
		this.code = code;
		this.startingPrice = price;
		this.price = price;
	}
	
	public static class StockInfoWarpper {
		public final Collection<StockInfo> stockPrices;

		public StockInfoWarpper(Collection<StockInfo> stockPrices) {
			this.stockPrices = Collections.unmodifiableCollection(stockPrices);
		}
	}
	
}
